### ISSUE TITLE
[P2] Broken division of negative numbers

### ISSUE THREAD

**Issue by ukasz** [tactcomplabs/rev#84]
We observe calculation errors on negative number division, see example below.
```
#define assert(x) if (!(x)) { asm(".byte 0x00"); asm(".byte 0x00"); asm(".byte 0x00"); asm(".byte 0x00"); }
int main(){
        int d = -20;
        int x = -3;
        int div = d/x;
        assert(div == 6);
        return 0;
}
```

**Comment by leekillough** [comment]
I cannot reproduce this. I [added a test](https://github.com/tactcomplabs/rev/pull/93).

**Comment by leekillough** [comment]
Fixed by #115
